backend/src/main/java/io/github/projektalmanac/amoxtli/backend/service/UserService.java

```diff
@@ -305,5 +311,89 @@ public SessionTokenDto iniciarSesion(CredencialesDto credenciales){ //hash passw
 
 
     }
+    /**/
+    public ValidaPuedeIntercambiar200ResponseDto validaIntercambio(Integer id) {
+
+        Optional<User> userOpt = userRepository.findById(id);
+        if (userOpt.isPresent()) {
+            throw new UserNotFoundException(id);
+        }
+        User user = userOpt.get();
+        if (!user.isVerifiedEmail()) {
+            throw new EmailUserNotVerificationException(id);
+        }
+
+        if(!user.getExchangesAccepting().isEmpty() && user.getExchangesAccepting().size() < 4){
+            throw new InvalidUserException("No puedes realizar más de 4 intercambios a la vez.");
+        }
+        ValidaPuedeIntercambiar200ResponseDto response = new ValidaPuedeIntercambiar200ResponseDto();
+        response.setPuedeIntercambiar(true);
+        response.mensaje("Puede solicitar intercambiar un libro.");
+
+        return response;
+    }
+
+    public IntercambioDto solicitaIntercambio(Integer idOfertante, CreacionIntercambioDto creacionIntercambioDto) {
+
+        if(creacionIntercambioDto == null){
+            throw new IllegalArgumentException("El cuerpo de la petición no puede ser nulo.");
+        }
+        // Recuperamos la información del usuario y del libro que recibe el intercambio
+        Integer idAceptante = creacionIntercambioDto.getIdAceptante();
+        Integer idLibroAceptante = creacionIntercambioDto.getIdLibroAceptante();
+
+        if (Objects.equals(idOfertante, idAceptante)){
+            throw new BadRequestException("No puedes intercambiar libros contigo mismo.");
+        }
+        // Obtenemos a los usuarios y al libro que se intercambiará
+        Optional <User> userOptionalOfertante = this.userRepository.findById(idOfertante);
+        if (userOptionalOfertante.isEmpty()) {
+            throw new UserNotFoundException(idOfertante);
+        }
```

Comment:
```suggestion
        User userOfertante = this.userRepository.findById(idOfertante).orElseThrow(
            () -> new UserNotFoundException(idOfertante)
        )
```

Podrías hacer esto en lugar de estar llamando `get` cada vez que quieres acceder al valor